### PR TITLE
121 performance tests

### PR DESCRIPTION
**Pull request by rjbrown2**
Just adding performance tests into dev, so that they can be used within code optimization efforts.

### CODE REVIEW

File: util/src_util/pt_auth_as.c [hunk 1]
```diff
@@ -0,0 +1,1204 @@
...
+#include "shared_util.h"
+#include <stdio.h>
+
+#include <time.h>
+#include <unistd.h>
+
+int num_frames_1K = 1000;
+int num_frames_100 = 100;
+
+void Write_To_File(uint16_t enc_frame_len, float total_time, char* test_name, int num_frames, int reset)
```
Comment (jlucas9) [inline util/src_util/pt_auth_as.c:38]: Should both `Write_To_File` and `Apply_Security_Loop` be moved to another file for reuse across the entire project?

File: util/src_util/pt_auth_as.c [hunk 2]
```diff
@@ -0,0 +1,1204 @@
...
+    char* data_b = NULL;
+    int data_l = 0;
+
+    uint8_t* ptr_enc_frame = NULL;
+    uint16_t enc_frame_len = 0;
+    SadbRoutine sadb_routine = get_sadb_routine_inmemory();
+
+    hex_conversion(data_h, &data_b, &data_l);
+
+    SecurityAssociation_t* test_association = malloc(sizeof(SecurityAssociation_t) * sizeof(uint8_t));
```
Comment (jlucas9) [inline util/src_util/pt_auth_as.c:116]: Does `test_association` get freed anywhere? May apply to multiple tests.

File: util/src_util/pt_gcry_base.c [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+
+#include <time.h>
+#include <unistd.h>
+
+#include "crypto.h"
+#include "crypto_error.h"
+
+#include "shared_util.h"
+
+void random_big_buffer(char* buffer, int32_t buffer_len)
```
Comment (jlucas9) [inline util/src_util/pt_gcry_base.c:38]: Should `random_big_buffer` be moved to another file for reuse across the entire project?

File: util/src_util/pt_auth_as.c (hunk 1 above)
Comment (rjbrown2) [reply util/src_util/pt_auth_as.c:38]: I don't think anything else will be using those functions.  They were just for my testing purposes.  If we so desire, I can move them, though.

File: util/src_util/pt_auth_as.c (hunk 2 above)
Comment (rjbrown2) [reply util/src_util/pt_auth_as.c:116]: Not within the unit tests, no

File: util/src_util/pt_gcry_base.c (hunk 3 above)
Comment (rjbrown2) [reply util/src_util/pt_gcry_base.c:38]: That's another one, that I was just creating for the purposes of these tests, but if so desired, we can indeed move it to a shared util.

**Review by IbraheemYSaleh — APPROVED** [review]
Looks fine to me, with or without John's suggested improvements.

**Pull request merged**